### ISSUE TITLE
cAdvisor on kubelet 1.3.6 improperly reporting memory usage

### ISSUE THREAD

**Issue by SleepyBrett** [kubernetes/kubernetes#32304]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.): memory

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Kubernetes version** (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.6+ae4550c", GitCommit:"ae4550cc9c89a593bcda6678df201db1b208133b", GitTreeState:"not a git tree", BuildDate:"2016-08-30T15:45:51Z", GoVersion:"go1.7", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.6", GitCommit:"ae4550cc9c89a593bcda6678df201db1b208133b", GitTreeState:"clean", BuildDate:"2016-08-26T18:06:06Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Cloud provider or hardware configuration**: AWS
- **OS** (e.g. from /etc/os-release): 
  NAME=CoreOS
  ID=coreos
  VERSION=1122.1.0
  VERSION_ID=1122.1.0
  BUILD_ID=2016-08-11-0252
  PRETTY_NAME="CoreOS 1122.1.0 (MoreOS)"
  ANSI_COLOR="1;32"
  HOME_URL="https://coreos.com/"
  BUG_REPORT_URL="https://github.com/coreos/bugs/issues"
- **Kernel** (e.g. `uname -a`):
  Linux xxxx 4.7.0-coreos #1 SMP Thu Aug 11 02:44:21 UTC 2016 x86_64 Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz GenuineIntel GNU/Linux
- **Install tools**:
- **Others**:

**What happened**:
cAdvisor reporting 0 memory usage for all containers

**What you expected to happen**:
cAdvisor to report actual memory usage statistics

**How to reproduce it** (as minimally and precisely as possible):
curl http://localhost:4194/api/v1.1/containers 

Observe that all memory blocks in the json look like this:
`"memory": {
        "usage": 0,
        "cache": 0,
        "rss": 0,
        "working_set": 0,
        "failcnt": 0,
        "container_data": {
          "pgfault": 0,
          "pgmajfault": 0
        },
        "hierarchical_data": {
          "pgfault": 0,
          "pgmajfault": 0
        }
      },`

**Anything else do we need to know**:
This seems to be a new behavior in 1.3.6

**Comment by SleepyBrett** [comment]
Initially all nodes in our cluster were effected, after restarting a kubelet that node would again start reporting stats correctly.

**Comment by SleepyBrett** [comment]
An update, we've had two nodes since the kubelet restart arrive in this state again a few hours apart. Nothing in the kubelet logs (looked back 10 minutes before it stopped reporting) but it happily mounting things.

**Comment by SleepyBrett** [comment]
Another update, this also stops the cpu metrics as well.

**Comment by hubt** [comment]
I believe I am encountering this in kubernetes 1.4 as well using debian/jessie and docker 1.11.2. It seems like updating to docker 1.12.1 helped. This is a significant problem for us because it means Heapster is reporting 0% cpu usage all the time and so hpas are broken.

**Comment by pieterlange** [comment]
I've been experiencing the same issue with coreos stable and k8s 1.4. Restarting kubelet on a node "works" for me too, for a couple of hours.

**Comment by gene-telligent** [comment]
Bumping and adding to the incidence rate:

**Version**

Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.5", GitCommit:"b0deb2eb8f4037421077f77cb163dbb4c0a2a9f5", GitTreeState:"clean", BuildDate:"2016-08-11T20:29:08Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.5", GitCommit:"b0deb2eb8f4037421077f77cb163dbb4c0a2a9f5", GitTreeState:"clean", BuildDate:"2016-08-11T20:21:58Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}

**Cloud provider or hardware configuration:** AWS
**OS (e.g. from /etc/os-release):**
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

** Kernel (e.g. uname -a): **
Linux ip-172-29-0-9 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-2+deb8u3 (2016-07-02) x86_64 GNU/Linux


The CPU reporting looks like this:
![screen shot 2016-10-26 at 4 39 03 pm](https://cloud.githubusercontent.com/assets/11700535/19749095/0d990c24-9b9b-11e6-9aa2-6670eba63f6a.png)

....with no records in the logs that anything strange has happened.

**Comment by agsmith** [comment]
Maybe related to an issue I've seen with the kubelet'c cAdvisor dying after 12hours or so:
https://github.com/kubernetes/kubernetes/issues/33192

**Comment by vishh** [comment]
@dashpole can you take a look at this issue?

**Comment by dashpole** [comment]
Ill look at it this week

**Comment by shamil** [comment]
I've similar problem #32414 

Only after restarting kubelet the metrics reporting are back to normal. I've upgraded to k8s 1.4 the issue still exist.

Do we even have a clue?

**Comment by dashpole** [comment]
Quick update: I have have a coreos stable instance that has been running cadvisor for more than 24 hours now, and is still reporting metrics...  I'll post an update if and when it stops reporting.

**Comment by dashpole** [comment]
I have had a coreos cluster up for over 2 weeks, and none of the nodes have stopped reporting

**Comment by dchen1107** [comment]
I didn't look into this in detail. But since I saw coreos is mentioned here, I guess you might hit this issue: https://github.com/kubernetes/kubernetes/issues/30939

**Comment by dashpole** [comment]
That does look like a possible cause, and would explain why I haven't been able to reproduce.  Thanks @dchen1107!  
@shamil and @SleepyBrett can you see if you are using systemd as your Cgroup Driver (run: docker info)?
During my testing, I have been using cgroupfs on coreos 1185.3.0 (MoreOS).

**Comment by hubt** [comment]
Is the conclusion that this is a docker problem that affects 1.11  with cgroupfs(which is the used in systemd >= 226).

Because I believe I am seeing this issue, but I am using docker 1.11.2 and systemd 215(k8s version 1.4.0 provisioned with kops). I am using the cgroupfs driver.

**Comment by shamil** [comment]
The cluster provisioned using `kops`

K8s version: `1.4.6`
Kops version: `1.4.1`
Systemd version: `215`
Docker version: `1.11.2`
Cgroup Driver: `cgroupfs`

**Comment by dashpole** [comment]
This may be fixed by [cadvisor#1573](https://github.com/google/cadvisor/pull/1573)

**Comment by dashpole** [comment]
According to [cadvisor#1573](https://github.com/google/cadvisor/pull/1573), it fixes [cadvisor#1572](https://github.com/google/cadvisor/issues/1572), which appears to be a duplicate of this issue.  This change will be included in the kubernetes 1.6 release.  I am closing this for now, but feel free to reopen if anyone runs into this problem after 1.6.
/close
